cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/efs/AwsEfsFileSystem.java

```diff
@@ -53,9 +55,17 @@
 
     private String throughputMode;
 
+    private List<String> associatedInstanceGroupNames;
+
     public AwsEfsFileSystem(String backupPolicyStatus, boolean encrypted, String fileSystemPolicy, Map<String, String> fileSystemTags, String kmsKeyId,
-            List<String> lifeCyclePolicies, String performanceMode, Double provisionedThroughputInMibps, String throughputMode) {
-        this.backupPolicyStatus = backupPolicyStatus != null ? backupPolicyStatus : BACKUP_POLICY_STATUS_DEFAULT;
+            List<String> lifeCyclePolicies, String performanceMode, Double provisionedThroughputInMibps, String throughputMode,
+            List<String> associatedInstanceGroupNames) {
+
+        if (associatedInstanceGroupNames == null || associatedInstanceGroupNames.size() == 0) {
+            throw new InvalidInputException("EFS associated instance group names are not set");
```

Comment:
I do not think it makes sense to throw an AWS exception in CB. Please use something more standard like `IllegalArgumentException` or whatever more CB specific.

Apart from the exception type, we shall have some proper validation already at the controller/converter/service level so that only sane values are passed down to `cloud-aws`.

Reply:
I was planning to do validation in the next jira CB-11225 to keep this one small. I will add validation in this one then.

Changed the exception type.